language/move-binary-format/src/compatibility.rs

```diff
@@ -49,36 +54,27 @@ impl Compatibility {
             check_struct_and_pub_function_linking: false,
             check_struct_layout: false,
             check_friend_linking: false,
-        }
-    }
-
-    pub fn new(
-        check_struct_and_pub_function_linking: bool,
-        check_struct_layout: bool,
-        check_friend_linking: bool,
-    ) -> Self {
-        Self {
-            check_struct_and_pub_function_linking,
-            check_struct_layout,
-            check_friend_linking,
+            check_private_entry_linking: false,
         }
     }
 
     pub fn need_check_compat(&self) -> bool {
         self.check_struct_and_pub_function_linking
             || self.check_friend_linking
             || self.check_struct_layout
+            || self.check_private_entry_linking
```

Comment:
nit: To avoid some boilerplate:

```suggestion
        self != &Self::no_check()
```